desc/objectives/_coils.py

```diff
@@ -597,3 +597,207 @@ def compute(self, params, constants=None):
+        Specify how to compute jacobian matrix, either forward mode or reverse mode AD.
+        "auto" selects forward or reverse mode based on the size of the input and output
+        of the objective. Has no effect on self.grad or self.hess which always use
+        reverse mode and forward over reverse mode respectively.
+    source_grid : Grid, optional
+        Collocation grid containing the nodes to evaluate field source at on
+        the winding surface. (used if e.g. field is a CoilSet or
+        FourierCurrentPotentialField). Defaults to the default for the
+        given field, see the docstring of the field object for the specific default.
+    eval_grid : Grid, optional
+        Collocation grid containing the nodes to evaluate the normal magnetic field at
+        plasma geometry at. Defaults to a LinearGrid(L=eq.L_grid, M=eq.M_grid,
+        zeta=jnp.array(0.0), NFP=eq.NFP).
+    name : str, optional
+        Name of the objective function.
+    """
+
+    _coordinates = "rtz"
+    _units = "T*m^2"
+    _print_value_fmt = "Toroidal Flux: {:10.3e} "
+
+    def __init__(
+        self,
+        eq,
+        field,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        loss_function=None,
+        deriv_mode="auto",
+        source_grid=None,
+        eval_grid=None,
+        name="toroidal-flux",
+    ):
+        if target is None and bounds is None:
+            target = eq.Psi
+        self._field = field
+        self._source_grid = source_grid
+        self._eval_grid = eval_grid
+        self._eq = eq
+
+        super().__init__(
+            things=[field],
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            loss_function=loss_function,
+            deriv_mode=deriv_mode,
+            name=name,
+        )
+
+    def build(self, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        eq = self._eq
+        if self._eval_grid is None:
+            eval_grid = LinearGrid(
+                L=eq.L_grid, M=eq.M_grid, zeta=jnp.array(0.0), NFP=eq.NFP
+            )
+            self._eval_grid = eval_grid
+        eval_grid = self._eval_grid
+
+        warnif(
+            not np.allclose(eval_grid.nodes[:, 2], eval_grid.nodes[0, 2]),
+            UserWarning,
+            "Evaluation grid should be at constant zeta",
+        )
+        if self._normalize:
+            self._normalization = eq.Psi
+
+        # ensure vacuum eq, as is unneeded for finite beta
+        pres = np.max(np.abs(eq.compute("p")["p"]))
+        curr = np.max(np.abs(eq.compute("current")["current"]))
+        errorif(
```

Comment:
i know i said erroring might be better here but I'm reconsidering.

Telling whether something is vacuum can be tricky because its always relative. IE pressure of 1e-8 might seem small, but if the whole problem is rescaled than it might be big. Similarly for current (especially since even if the loop integral of B is 0 to machine precision, when we divide by mu0 to get current its usually at least ~1e-8)